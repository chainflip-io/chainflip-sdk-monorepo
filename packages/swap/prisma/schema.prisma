// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

// `pnpm prisma format` after changes :)

generator client {
  provider        = "prisma-client-js"
  // https://github.com/prisma/prisma/issues/2584#issuecomment-646414549
  // https://github.com/prisma/prisma/issues/6603#issuecomment-849979849
  output          = "../node_modules/.prisma/client"
  previewFeatures = ["multiSchema"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  schemas  = ["public", "private"]
}

enum Chain {
  Polkadot
  Ethereum
  Bitcoin

  @@schema("public")
}

enum Asset {
  FLIP
  USDC
  DOT
  ETH
  BTC

  @@schema("public")
}

model SwapDepositChannel {
  id                    BigInt   @id @default(autoincrement())
  uuid                  String   @unique @default(uuid())
  srcAsset              Asset
  depositAddress        String
  expectedDepositAmount Decimal  @db.Decimal(30, 0)
  destAsset             Asset
  destAddress           String
  expiryBlock           Int
  issuedBlock           Int
  createdAt             DateTime @default(now())
  swaps                 Swap[]

  @@index([depositAddress])
  @@schema("public")
}

model Swap {
  id                        BigInt              @id @default(autoincrement())
  /// the swap id on the state chain
  nativeId                  BigInt              @unique
  srcAsset                  Asset
  depositAmount             Decimal              @db.Decimal(30, 0)
  intermediateAmount        Decimal?             @db.Decimal(30, 0)
  egressAmount              Decimal?             @db.Decimal(30, 0)
  destAsset                 Asset
  destAddress               String
  depositReceivedAt         DateTime
  depositReceivedBlockIndex String
  swapExecutedAt            DateTime?
  swapExecutedBlockIndex    String?
  egressCompletedAt         DateTime?
  egressCompletedBlockIndex String?
  swapDepositChannel        SwapDepositChannel? @relation(fields: [swapDepositChannelId], references: [id])
  swapDepositChannelId      BigInt?
  txHash                    String?             @unique
  egress                    Egress?
  createdAt                 DateTime            @default(now())
  updatedAt                 DateTime            @default(now()) @updatedAt

  @@schema("public")
}

model Egress {
  id          BigInt     @id @default(autoincrement())
  /// the egress id on the state chain
  nativeId    BigInt
  /// the swap id on the state chain
  chain       Chain
  amount      Decimal    @db.Decimal(30, 0)
  timestamp   DateTime
  swap        Swap?      @relation(fields: [swapId], references: [id])
  swapId      BigInt?    @unique
  broadcast   Broadcast? @relation(fields: [broadcastId], references: [id])
  broadcastId BigInt?
  createdAt   DateTime   @default(now())

  @@unique([nativeId, chain])
  @@schema("public")
}

model Broadcast {
  id       BigInt   @id @default(autoincrement())
  /// the broadcast id on the state chain
  nativeId BigInt
  chain    Chain
  egresses Egress[]

  @@unique([nativeId, chain])
  @@schema("public")
}

model State {
  id     Int @id @default(autoincrement())
  height Int @default(0)

  @@schema("private")
}

model MarketMaker {
  id        Int    @id @default(autoincrement())
  name      String @unique
  publicKey String

  @@schema("private")
}

enum ThirdPartyProtocol {
  lifi
  squid

  @@schema("public")
}

model ThirdPartySwap {
  id            BigInt             @id @default(autoincrement())
  uuid          String             @unique
  protocol      ThirdPartyProtocol
  routeResponse Json
  txHash        String
  txLink        String
  createdAt     DateTime           @default(now())

  @@schema("public")
}
