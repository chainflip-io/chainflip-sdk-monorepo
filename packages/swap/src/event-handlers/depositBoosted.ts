import { arbitrumIngressEgressDepositBoosted as arbitrumSchema160 } from '@chainflip/processor/160/arbitrumIngressEgress/depositBoosted';
import { bitcoinIngressEgressDepositBoosted as bitcoinSchema160 } from '@chainflip/processor/160/bitcoinIngressEgress/depositBoosted';
import { ethereumIngressEgressDepositBoosted as ethereumSchema160 } from '@chainflip/processor/160/ethereumIngressEgress/depositBoosted';
import { polkadotIngressEgressDepositBoosted as polkadotSchema160 } from '@chainflip/processor/160/polkadotIngressEgress/depositBoosted';
import { solanaIngressEgressDepositBoosted as solanaSchema160 } from '@chainflip/processor/160/solanaIngressEgress/depositBoosted';
import { bitcoinIngressEgressDepositBoosted as bitcoinSchema170 } from '@chainflip/processor/170/bitcoinIngressEgress/depositBoosted';
import { arbitrumIngressEgressDepositBoosted as arbitrumSchema180 } from '@chainflip/processor/180/arbitrumIngressEgress/depositBoosted';
import { bitcoinIngressEgressDepositBoosted as bitcoinSchema180 } from '@chainflip/processor/180/bitcoinIngressEgress/depositBoosted';
import { ethereumIngressEgressDepositBoosted as ethereumSchema180 } from '@chainflip/processor/180/ethereumIngressEgress/depositBoosted';
import { polkadotIngressEgressDepositBoosted as polkadotSchema180 } from '@chainflip/processor/180/polkadotIngressEgress/depositBoosted';
import { solanaIngressEgressDepositBoosted as solanaSchema180 } from '@chainflip/processor/180/solanaIngressEgress/depositBoosted';
import { z } from 'zod';
import { assetConstants } from '@/shared/enums';
import { ONE_IN_PIP } from '@/shared/functions';
import { SwapFeeType } from '@/swap/client';
import { getDepositTxRef } from '@/swap/event-handlers/common';
import { EventHandlerArgs } from '.';

const arbitrumSchema = z.union([arbitrumSchema160, arbitrumSchema180]);
const bitcoinSchema = z.union([bitcoinSchema160, bitcoinSchema170, bitcoinSchema180]);
const ethereumSchema = z.union([ethereumSchema160, ethereumSchema180]);
const polkadotSchema = z.union([polkadotSchema160, polkadotSchema180]);
const solanaSchema = z.union([solanaSchema160, solanaSchema180]);

const depositBoostedSchema = z.union([
  arbitrumSchema,
  bitcoinSchema,
  ethereumSchema,
  polkadotSchema,
  solanaSchema,
]);

export type DepositBoostedArgs = z.input<typeof depositBoostedSchema>;

// DepositBoosted event is emitted instead of DepositFinalised event in v140 due to boost
// We need to update the depositAmount and store the ingress fee just like we do in the DepositFinalised event
export const depositBoosted = async ({ prisma, event, block }: EventHandlerArgs) => {
  const {
    asset,
    boostFee,
    action,
    ingressFee,
    amounts,
    prewitnessedDepositId,
    blockHeight,
    ...rest
  } = depositBoostedSchema.parse(event.args);
  const depositDetails = 'depositDetails' in rest ? rest.depositDetails : undefined;
  const maxBoostFeeBps = 'maxBoostFeeBps' in rest ? rest.maxBoostFeeBps : undefined;

  const txRef = getDepositTxRef(assetConstants[asset].chain, depositDetails, blockHeight);

  if (action.__kind === 'Swap' || action.__kind === 'CcmTransfer') {
    const depositAmount = amounts.reduce((acc, [, amount]) => acc + amount, BigInt(0));
    const effectiveBoostFeeBps = (boostFee * BigInt(ONE_IN_PIP)) / BigInt(depositAmount.toString());

    const fees = {
      create: [
        { type: SwapFeeType.BOOST, asset, amount: boostFee.toString() },
        { type: SwapFeeType.INGRESS, asset, amount: ingressFee.toString() },
      ],
    };

    const nativeId = action.swapRequestId;
    const data = {
      maxBoostFeeBps,
      effectiveBoostFeeBps: Number(effectiveBoostFeeBps),
      prewitnessedDepositId,
      depositTransactionRef: txRef,
      depositBoostedAt: new Date(block.timestamp),
      depositBoostedBlockIndex: `${block.height}-${event.indexInBlock}`,
      fees,
    };

    await prisma.swapRequest.update({ where: { nativeId }, data });
  }
};
