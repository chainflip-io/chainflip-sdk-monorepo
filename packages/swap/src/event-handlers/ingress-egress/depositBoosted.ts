import { arbitrumIngressEgressDepositBoosted as arbitrumSchema180 } from '@chainflip/processor/180/arbitrumIngressEgress/depositBoosted';
import { bitcoinIngressEgressDepositBoosted as bitcoinSchema180 } from '@chainflip/processor/180/bitcoinIngressEgress/depositBoosted';
import { ethereumIngressEgressDepositBoosted as ethereumSchema180 } from '@chainflip/processor/180/ethereumIngressEgress/depositBoosted';
import { polkadotIngressEgressDepositBoosted as polkadotSchema180 } from '@chainflip/processor/180/polkadotIngressEgress/depositBoosted';
import { solanaIngressEgressDepositBoosted as solanaSchema180 } from '@chainflip/processor/180/solanaIngressEgress/depositBoosted';
import { arbitrumIngressEgressDepositBoosted as arbitrumSchema190 } from '@chainflip/processor/190/arbitrumIngressEgress/depositBoosted';
import { assethubIngressEgressDepositBoosted as assethubSchema190 } from '@chainflip/processor/190/assethubIngressEgress/depositBoosted';
import { bitcoinIngressEgressDepositBoosted as bitcoinSchema190 } from '@chainflip/processor/190/bitcoinIngressEgress/depositBoosted';
import { ethereumIngressEgressDepositBoosted as ethereumSchema190 } from '@chainflip/processor/190/ethereumIngressEgress/depositBoosted';
import { polkadotIngressEgressDepositBoosted as polkadotSchema190 } from '@chainflip/processor/190/polkadotIngressEgress/depositBoosted';
import { solanaIngressEgressDepositBoosted as solanaSchema190 } from '@chainflip/processor/190/solanaIngressEgress/depositBoosted';
import { ChainflipChain } from '@chainflip/utils/chainflip';
import { z } from 'zod';
import { ONE_IN_PIP } from '@/shared/functions.js';
import { Prisma, SwapFeeType } from '../../client.js';
import { getDepositTxRef } from '../common.js';
import { EventHandlerArgs } from '../index.js';

const arbitrumSchema = z.union([arbitrumSchema190, arbitrumSchema180]).transform((args) => ({
  ...args,
  depositDetails: { chain: 'Arbitrum' as const, data: args.depositDetails },
}));
const bitcoinSchema = z.union([bitcoinSchema190, bitcoinSchema180]).transform((args) => ({
  ...args,
  depositDetails: { chain: 'Bitcoin' as const, data: args.depositDetails },
}));
const ethereumSchema = z.union([ethereumSchema190, ethereumSchema180]).transform((args) => ({
  ...args,
  depositDetails: { chain: 'Ethereum' as const, data: args.depositDetails },
}));
const polkadotSchema = z.union([polkadotSchema190, polkadotSchema180]).transform((args) => ({
  ...args,
  depositDetails: { chain: 'Polkadot' as const, data: args.depositDetails },
}));
const solanaSchema = z.union([solanaSchema190, solanaSchema180]).transform((args) => ({
  ...args,
  depositDetails: { chain: 'Solana' as const, data: undefined },
}));
const assethubSchema = assethubSchema190.transform((args) => ({
  ...args,
  depositDetails: { chain: 'Assethub' as const, data: args.depositDetails },
}));

const schemas = {
  Arbitrum: arbitrumSchema,
  Bitcoin: bitcoinSchema,
  Ethereum: ethereumSchema,
  Polkadot: polkadotSchema,
  Solana: solanaSchema,
  Assethub: assethubSchema,
} as const satisfies Record<ChainflipChain, z.ZodTypeAny>;

export type DepositBoostedArgsMap = {
  [C in ChainflipChain]: z.input<(typeof schemas)[C]>;
};

// DepositBoosted event is emitted instead of DepositFinalised event in v140 due to boost
// We need to update the depositAmount and store the ingress fee just like we do in the DepositFinalised event
export const depositBoosted =
  (chain: ChainflipChain) =>
  async ({ prisma, event, block }: EventHandlerArgs) => {
    const {
      asset,
      boostFee,
      action,
      ingressFee,
      amounts,
      prewitnessedDepositId,
      blockHeight,
      depositDetails,
      maxBoostFeeBps,
    } = schemas[chain].parse(event.args);

    const txRef = getDepositTxRef(depositDetails, blockHeight);

    if (action.__kind === 'Swap' || action.__kind === 'CcmTransfer') {
      const depositAmount = amounts.reduce((acc, [, amount]) => acc + amount, BigInt(0));
      const effectiveBoostFeeBps =
        (boostFee * BigInt(ONE_IN_PIP)) / BigInt(depositAmount.toString());

      const fees = {
        create: [
          { type: SwapFeeType.BOOST, asset, amount: boostFee.toString() },
          { type: SwapFeeType.INGRESS, asset, amount: ingressFee.toString() },
        ],
      };

      const nativeId = action.swapRequestId;
      const data: Prisma.SwapRequestUpdateInput = {
        maxBoostFeeBps,
        effectiveBoostFeeBps: Number(effectiveBoostFeeBps),
        prewitnessedDepositId,
        depositTransactionRef: txRef,
        depositBoostedAt: new Date(block.timestamp),
        depositBoostedBlockIndex: `${block.height}-${event.indexInBlock}`,
        fees,
        depositAmount: depositAmount.toString(),
      };

      await prisma.swapRequest.update({ where: { nativeId }, data });
    }
  };
