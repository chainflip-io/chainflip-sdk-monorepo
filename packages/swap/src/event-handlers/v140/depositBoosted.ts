import { arbitrumIngressEgressDepositBoosted } from '@chainflip/processor/160/arbitrumIngressEgress/depositBoosted';
import { bitcoinIngressEgressDepositBoosted } from '@chainflip/processor/160/bitcoinIngressEgress/depositBoosted';
import { ethereumIngressEgressDepositBoosted } from '@chainflip/processor/160/ethereumIngressEgress/depositBoosted';
import { polkadotIngressEgressDepositBoosted } from '@chainflip/processor/160/polkadotIngressEgress/depositBoosted';
import { solanaIngressEgressDepositBoosted } from '@chainflip/processor/160/solanaIngressEgress/depositBoosted';
import { z } from 'zod';
import { ONE_IN_PIP } from '@/shared/functions';
import { SwapFeeType } from '@/swap/client';
import { EventHandlerArgs } from '..';

const arbitrumSchema = arbitrumIngressEgressDepositBoosted;
const bitcoinSchema = bitcoinIngressEgressDepositBoosted;
const ethereumSchema = ethereumIngressEgressDepositBoosted;
const polkadotSchema = polkadotIngressEgressDepositBoosted;
const solanaSchema = solanaIngressEgressDepositBoosted;

const depositBoostedSchema = z.union([
  arbitrumSchema,
  bitcoinSchema,
  ethereumSchema,
  polkadotSchema,
  solanaSchema,
]);

export type DepositBoostedArgs = z.input<typeof depositBoostedSchema>;

// DepositBoosted event is emitted instead of DepositFinalised event in v140 due to boost
// We need to update the depositAmount and store the ingress fee just like we do in the DepositFinalised event
export const depositBoosted = async ({ prisma, event, block }: EventHandlerArgs) => {
  const { asset, boostFee, action, ingressFee, amounts, prewitnessedDepositId } =
    depositBoostedSchema.parse(event.args);

  if (action.__kind === 'Swap') {
    const depositAmount = amounts.reduce((acc, [, amount]) => acc + amount, BigInt(0));
    const effectiveBoostFeeBps = (boostFee * BigInt(ONE_IN_PIP)) / BigInt(depositAmount.toString());

    const fees = {
      create: [
        { type: SwapFeeType.BOOST, asset, amount: boostFee.toString() },
        { type: SwapFeeType.INGRESS, asset, amount: ingressFee.toString() },
      ],
    };

    const nativeId = action.swapRequestId;
    const data = {
      effectiveBoostFeeBps: Number(effectiveBoostFeeBps),
      prewitnessedDepositId,
      depositBoostedAt: new Date(block.timestamp),
      depositBoostedBlockIndex: `${block.height}-${event.indexInBlock}`,
      fees,
    };

    await prisma.swapRequest.update({ where: { nativeId }, data });
  }
};
