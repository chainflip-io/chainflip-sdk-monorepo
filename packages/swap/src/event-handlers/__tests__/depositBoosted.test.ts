import { describe, it, expect, beforeEach } from 'vitest';
import prisma from '../../client';
import { depositBoosted, DepositBoostedArgs } from '../depositBoosted';

export const depositBoostedBtcMock = async ({
  action = { __kind: 'Swap', swapRequestId: '1' },
  amounts,
  channelId,
}: {
  action?: DepositBoostedArgs['action'];
  amounts?: [[number, string]];
  channelId?: string;
} = {}) => {
  const args: DepositBoostedArgs = {
    blockHeight: 120,
    asset: {
      __kind: 'Btc',
    },
    amounts: amounts ?? [[5, '1000000']],
    prewitnessedDepositId: '101',
    channelId: channelId ?? '1',
    ingressFee: '1000',
    boostFee: '500',
    action,
    depositAddress: {
      value: '0x52890cc3438775253262c88df4ab47841581ac04',
      __kind: 'P2PKH',
    },
    depositDetails: {
      txId: '0x626b620f866caa7474598d3a34a752dba98e5c55f1e3de1c310b75ad093b32c7',
      vout: 0,
    },
  };

  if (action.__kind === 'Swap' || action.__kind === 'CcmTransfer') {
    await prisma.swapRequest.create({
      data: {
        nativeId: BigInt(action.swapRequestId),
        srcAsset: 'Btc',
        destAsset: 'Flip',
        originType: 'VAULT',
        requestType: 'LEGACY_SWAP',
        depositAmount: '1000000',
        swapInputAmount: '1000000',
        swapRequestedAt: new Date('2023-01-01T00:00:00.000Z'),
        swapRequestedBlockIndex: '92-398',
      },
    });
  }

  return {
    block: {
      height: 120,
      timestamp: 1670337105000,
      hash: '0x123',
      specId: 'test@150',
    },
    event: {
      args,
      name: 'BitcoinIngressEgress.DepositBoosted',
      indexInBlock: 7,
    },
  } as const;
};

describe('depositBoosted', () => {
  beforeEach(async () => {
    await prisma.$queryRaw`TRUNCATE TABLE "private"."DepositChannel" CASCADE`;
    await prisma.$queryRaw`TRUNCATE TABLE "SwapDepositChannel", "Swap", "SwapRequest" CASCADE`;
  });

  it('updates the values for an existing swap', async () => {
    const { event, block } = await depositBoostedBtcMock({ amounts: [[5, '1000000']] });

    await depositBoosted({ prisma, event, block });

    const request = await prisma.swapRequest.findFirstOrThrow({
      include: { fees: { select: { asset: true, amount: true, type: true } } },
    });

    expect(request).toMatchSnapshot({
      id: expect.any(BigInt),
    });
  });

  it('updates the values for an existing ccm swap', async () => {
    const { event, block } = await depositBoostedBtcMock({
      action: { __kind: 'CcmTransfer', swapRequestId: '1' },
      amounts: [[5, '1000000']],
    });

    await depositBoosted({ prisma, event, block });

    const request = await prisma.swapRequest.findFirstOrThrow({
      include: { fees: { select: { asset: true, amount: true, type: true } } },
    });

    expect(request).toMatchSnapshot({
      id: expect.any(BigInt),
    });
  });

  it('updates the values for event with 180 schema', async () => {
    await prisma.swapRequest.create({
      data: {
        nativeId: 159,
        srcAsset: 'Btc',
        destAsset: 'Eth',
        destAddress: '0x41ad2bc63a2059f9b623533d87fe99887d794847',
        swapInputAmount: '10000000',
        originType: 'VAULT',
        requestType: 'REGULAR',
        swapRequestedAt: new Date('2024-08-06T00:00:00.000Z'),
      },
    });

    await depositBoosted({
      prisma,
      block: {
        height: 120,
        timestamp: 1670337105000,
        hash: '0x123',
        specId: 'test@150',
      },
      event: {
        args: {
          asset: {
            __kind: 'Btc',
          },
          action: {
            __kind: 'Swap',
            swapRequestId: '159',
          },
          amounts: [[4, '10000000']],
          boostFee: '4000',
          channelId: '7',
          ingressFee: '79',
          originType: {
            __kind: 'DepositChannel',
          },
          blockHeight: '49020',
          depositAddress: {
            value: '0x6637a004f3e8536158ee981490b559d6ea0ae92f086d4911454ab88fdbc2c523',
            __kind: 'Taproot',
          },
          depositDetails: {
            id: {
              txId: '0xcc76691987ad74f75e247bf86549d643f281ff1bd2256e994c899a549d6809a6',
              vout: 0,
            },
            amount: '10000000',
            depositAddress: {
              pubkeyX: '0xd02106f2d8b619bc95b662340f904af6d67bd0efebd1692a839cbc172bbd8a07',
              scriptPath: {
                salt: 7,
                tapleafHash: '0x5c2c1bf4030a5249bb8911aa6ee785f4969764ed6f1752adfbf643ffd011808d',
                unlockScript: {
                  bytes:
                    '0x577520d02106f2d8b619bc95b662340f904af6d67bd0efebd1692a839cbc172bbd8a07ac',
                },
                tweakedPubkeyBytes:
                  '0x026637a004f3e8536158ee981490b559d6ea0ae92f086d4911454ab88fdbc2c523',
              },
            },
          },
          maxBoostFeeBps: 7,
          prewitnessedDepositId: '6',
        },
        name: 'BitcoinIngressEgress.DepositBoosted',
        indexInBlock: 7,
      },
    });

    const request = await prisma.swapRequest.findFirstOrThrow({
      include: { fees: { select: { asset: true, amount: true, type: true } } },
    });

    expect(request).toMatchSnapshot({
      id: expect.any(BigInt),
    });
  });
});
