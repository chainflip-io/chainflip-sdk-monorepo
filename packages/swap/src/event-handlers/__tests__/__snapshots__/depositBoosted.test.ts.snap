// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`depositBoosted > updates the values for an existing ccm swap 1`] = `
{
  "ccmDepositReceivedBlockIndex": null,
  "ccmGasBudget": null,
  "ccmMessage": null,
  "completedAt": null,
  "completedBlockIndex": null,
  "dcaChunkIntervalBlocks": null,
  "dcaNumberOfChunks": null,
  "depositAmount": "1000000",
  "depositBoostedAt": 2022-12-06T14:31:45.000Z,
  "depositBoostedBlockIndex": "120-7",
  "depositFinalisedAt": null,
  "depositFinalisedBlockIndex": null,
  "depositTransactionRef": "c7323b09ad750b311cdee3f1555c8ea9db52a7343a8d597474aa6c860f626b62",
  "destAddress": null,
  "destAsset": "Flip",
  "effectiveBoostFeeBps": 5,
  "egressId": null,
  "fees": [
    {
      "amount": "500",
      "asset": "Btc",
      "type": "BOOST",
    },
    {
      "amount": "1000",
      "asset": "Btc",
      "type": "INGRESS",
    },
  ],
  "fokMinPriceX128": null,
  "fokRefundAddress": null,
  "fokRetryDurationBlocks": null,
  "id": Any<BigInt>,
  "maxBoostFeeBps": 0,
  "nativeId": 1n,
  "originType": "VAULT",
  "prewitnessedDepositId": 101n,
  "refundEgressId": null,
  "requestType": "LEGACY_SWAP",
  "srcAddress": null,
  "srcAsset": "Btc",
  "swapDepositChannelId": null,
  "swapInputAmount": "1000000",
  "swapOutputAmount": null,
  "swapRequestedAt": 2023-01-01T00:00:00.000Z,
  "swapRequestedBlockIndex": "92-398",
  "totalBrokerCommissionBps": 0,
}
`;

exports[`depositBoosted > updates the values for an existing swap 1`] = `
{
  "ccmDepositReceivedBlockIndex": null,
  "ccmGasBudget": null,
  "ccmMessage": null,
  "completedAt": null,
  "completedBlockIndex": null,
  "dcaChunkIntervalBlocks": null,
  "dcaNumberOfChunks": null,
  "depositAmount": "1000000",
  "depositBoostedAt": 2022-12-06T14:31:45.000Z,
  "depositBoostedBlockIndex": "120-7",
  "depositFinalisedAt": null,
  "depositFinalisedBlockIndex": null,
  "depositTransactionRef": "c7323b09ad750b311cdee3f1555c8ea9db52a7343a8d597474aa6c860f626b62",
  "destAddress": null,
  "destAsset": "Flip",
  "effectiveBoostFeeBps": 5,
  "egressId": null,
  "fees": [
    {
      "amount": "500",
      "asset": "Btc",
      "type": "BOOST",
    },
    {
      "amount": "1000",
      "asset": "Btc",
      "type": "INGRESS",
    },
  ],
  "fokMinPriceX128": null,
  "fokRefundAddress": null,
  "fokRetryDurationBlocks": null,
  "id": Any<BigInt>,
  "maxBoostFeeBps": 0,
  "nativeId": 1n,
  "originType": "VAULT",
  "prewitnessedDepositId": 101n,
  "refundEgressId": null,
  "requestType": "LEGACY_SWAP",
  "srcAddress": null,
  "srcAsset": "Btc",
  "swapDepositChannelId": null,
  "swapInputAmount": "1000000",
  "swapOutputAmount": null,
  "swapRequestedAt": 2023-01-01T00:00:00.000Z,
  "swapRequestedBlockIndex": "92-398",
  "totalBrokerCommissionBps": 0,
}
`;

exports[`depositBoosted > updates the values for event with 180 schema 1`] = `
{
  "ccmDepositReceivedBlockIndex": null,
  "ccmGasBudget": null,
  "ccmMessage": null,
  "completedAt": null,
  "completedBlockIndex": null,
  "dcaChunkIntervalBlocks": null,
  "dcaNumberOfChunks": null,
  "depositAmount": null,
  "depositBoostedAt": 2022-12-06T14:31:45.000Z,
  "depositBoostedBlockIndex": "120-7",
  "depositFinalisedAt": null,
  "depositFinalisedBlockIndex": null,
  "depositTransactionRef": "a609689d549a894c996e25d21bff81f243d64965f87b245ef774ad87196976cc",
  "destAddress": "0x41ad2bc63a2059f9b623533d87fe99887d794847",
  "destAsset": "Eth",
  "effectiveBoostFeeBps": 4,
  "egressId": null,
  "fees": [
    {
      "amount": "4000",
      "asset": "Btc",
      "type": "BOOST",
    },
    {
      "amount": "79",
      "asset": "Btc",
      "type": "INGRESS",
    },
  ],
  "fokMinPriceX128": null,
  "fokRefundAddress": null,
  "fokRetryDurationBlocks": null,
  "id": Any<BigInt>,
  "maxBoostFeeBps": 7,
  "nativeId": 159n,
  "originType": "VAULT",
  "prewitnessedDepositId": 6n,
  "refundEgressId": null,
  "requestType": "REGULAR",
  "srcAddress": null,
  "srcAsset": "Btc",
  "swapDepositChannelId": null,
  "swapInputAmount": "10000000",
  "swapOutputAmount": null,
  "swapRequestedAt": 2024-08-06T00:00:00.000Z,
  "swapRequestedBlockIndex": null,
  "totalBrokerCommissionBps": 0,
}
`;
